Model/DutyCalculator.php

```diff
@@ -116,84 +116,333 @@ public function __construct(
+     * @param Magento\Quote\Api\Data\AddressInterface $address
+     * @param bool $apply  - whether duty is applicable or not
+     *
+     */
+    public function handleTaxApplicability($address, $apply)
+    {
+        $this->_logger->debug('In handleTaxApplicability method');
+        $this->_logger->debug($this->checkoutSession->getReachDuty());
+        $this->_logger->debug($address->getCountryId());
+        $this->_logger->debug($address->getRegionCode());
+        $this->_logger->debug($this->checkoutSession->getPrevCountry());
+        $this->_logger->debug($this->checkoutSession->getPrevRegion());
+        $this->_logger->debug($apply);
+        $quote = $this->checkoutSession->getQuote();
+
+        if ($apply || !$this->getIsOptional($address->getCountryId())) {
+           $quote->setReachDuty($this->checkoutSession->getReachDuty());
+           $this->response->setIsOptional($this->getIsOptional($address->getCountryId()));
+           $this->response->setSuccess(true);
+           $this->response->setDuty($this->checkoutSession->getRechDuty());
+           $this->_logger->debug('In handleTaxApplicability method --- apply is true');
+        }
+        else {
+           $this->response->setIsOptional($this->getIsOptional($address->getCountryId()));
+           $this->response->setSuccess(true);
+           //as apply duty and tax(DT) is not selected duty and tax would not be used in total pricing
+           //so setting all relevant quote values to zero
+           //$quote->setDuty(0) ;
+           $quote->setReachDuty(0) ;
+           $this->response->setDuty($this->checkoutSession->getReachDuty()); //the DT checkbox and label  should still appear
+           //if this amount is more than 0 even though the user did not choose to apply it on
+           //computation of total landed cost/billing
+           $this->_logger->debug('In handleTaxApplicability method --- apply is false');
+        }
+
+        $quote->save();//this is needed so that different aspects related to a quote are available on other pages.
+        //More specifically saving quote in database is needed as apparently Magento session scopes are not application
+        // wide but code area specific (?)
+    }
+
+
+    /** handles when duty is not applicable (based on some configuration) for the country
+     * or shipping is not allowed for the country
+     * and saves proper state of quote so that the information is available beyond this page/area
+     * @param float $duty
+     * @param \Magento\Quote\Model\Quote $quote
+     */
+    public function handleNoDutyOrShippingCase($duty, $quote)
+    {
+        $this->response->setSuccess(true);
+        $this->response->setDuty($duty);
+        $this->checkoutSession->setReachDuty($duty);
+        $quote->setReachDuty($duty) ;
+        $this->_logger->debug('In duty or shipping not allowed section');
+        $quote->save();
+    }
+
+    /** Sets appropriate response (eventually be https response) values
```

Comment:
do we have a ticket to make this response https? If there's work we want to do my personal preference would be to make the ticket and potentially refer to it in the comment

Reply:
No ticket is necessary @lizzynibali. Magento is already tackling that. 
For example there are many types of response variables - even in dutyCalculator we have two types.
As for example \Reach\Payment\Api\Data\DutyResponseInterface that is also called response in the preexisting code (and I kept it the same) and then Magento\Framework\App\Response that is the HTTP response (if I am recalling it right). I added datatypes as part of @params to indicate which is which.
The response there is of  DutyResponseInterface category instead of the other one.
Would removing the  s from https or removing the comment altogether be more helpful to alleviate the confusion? Please let me know. I know I do not always express myself well enough. Thanks.

Reply:
Commit 87a79f49461544e2dea9c01befe0a2a40ff34b0c would hopefully help. In the comment mentioned what type of response it is (immediately at that point the response is of type DutyResponseInterface). Removed the part that indicates what happens in general during transfer over internet. Thanks. (No new ticket is necessary).